bin/docker-pull

```diff
@@ -10,5 +10,9 @@ else
 	eval $(aws ecr get-login --region $AWS_DEFAULT_REGION --registry-ids ${AWS_ECR_ACCOUNT_ID})
 fi
 
-# this should exit 0 even if the image is not there. For example, on a first run
-docker pull ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:latest || true
+# Warm the local docker cache (aka the local docker images) by pulling the previous commit and the images
+# tagged with the current branch, if they exist.
+# This should exit 0 even if the image is not there. For example, on a first run
+PREVIOUS_COMMIT=$(git rev-parse HEAD~1)
+docker pull ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:$PREVIOUS_COMMIT || true
+docker pull ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:$CI_BRANCH || true
```

Comment:
There is a chance (depending on docker file) that the cache can cause issues if the source changes but the cache includes an old version. It seems beneficial at least to pull the latest of this branch. 

I'm not sure the chance that the previous commit would have an image available. The time to fetch and check may not be an issue.